### PR TITLE
Hot fix for reshare issue

### PR DESCRIPTION
**Pull request by JesseAbram**
Currently reshare on testnet is failing due to a validation error that data is stale. This lines up the block number generated for reshare and the block number expected by the tss.

Regression test is difficult and needs more thinking.

### CODE REVIEW

**Comment by HCastano** [comment]
Can you make sure you test the runtime upgrade from `v0.3.0` -> `v0.3.1` on a locally running network first?

If you have time you could even do a dry run on a remote devnet. @entropyxyz/system-reliability-engineers may need to help you out with that if you haven't walked through the process before.

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -208,7 +208,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     // We update this if the runtime behaviour has changed. When this happens we set the
     // `impl_version` to `0`.
     #[allow(clippy::zero_prefixed_literal)]
-    spec_version: 00_03_00,
+    spec_version: 00_04_00,
```
Comment (HCastano) [inline runtime/src/lib.rs:211]:
```suggestion
    spec_version: 00_03_01,
```

**Comment by HCastano** [comment]
Oh I guess another consideration here actually is whether we want to run the full release pipeline and publish a `v0.3.1` release 🤔

**Comment by JesseAbram** [comment]
> Can you make sure you test the runtime upgrade from `v0.3.0` -> `v0.3.1` on a locally running network first?
> 
> If you have time you could even do a dry run on a remote devnet. @entropyxyz/system-reliability-engineers may need to help you out with that if you haven't walked through the process before.

yup done and also did that with chopsticks a few weeks ago

**Pull request closed without merging**